corehq/motech/dhis2/static/dhis2/js/dataset_map.js

```diff
@@ -0,0 +1,44 @@
+hqDefine("dhis2/js/dataset_map", [
+    "jquery",
+    "underscore",
+    "hqwebapp/js/initial_page_data",
+    "hqwebapp/js/crud_paginated_list_init",
+    "hqwebapp/js/widgets",
+], function (
+    $,
+    _,
+    initialPageData
+) {
+    $(function () {
+        var $sendNowResult = $('#send-now-result');
+
+        var handleSuccess = function (response) {
+            $sendNowResult.removeClass("hide text-danger text-success");
+            if (response.success) {
+                $sendNowResult.addClass("text-success");
+            } else {
+                $sendNowResult.addClass("text-danger");
+            }
+            $sendNowResult.text(gettext('DHIS2 response: ') + response.text);
+        };
+
+        var handleFailure = function (resp) {
+            $sendNowResult
+                .removeClass("hide text-success")
+                .addClass("text-danger");
+            $sendNowResult.text(
+                gettext('CommCare HQ was unable to send the DataSet: ')
+                + (resp.responseJSON ? resp.responseJSON['error'] : resp.statusText)
+            );
+        };
+
+        sendNow = function (dataSetMapId) {
```

Comment:
This is annoying me. The "Send Now" button needs to call this "sendNow" function. Here are the options I see:
* Use a global variable: Not great.
* Create a tiny viewmodel for this function, and bind it to the paginated list: Does not work, because there already is a viewmodel bound to the paginated list, and you can't bind more than one.
* Replace the definition of the [paginatedListModel](https://github.com/dimagi/commcare-hq/blob/ca162bb6e93b84c1e1a45cdf6fe6f4d4fc6df31c/corehq/apps/hqwebapp/static/hqwebapp/js/crud_paginated_list_init.js#L12) viewmodel: This seems to be [the approach that billing_statements.js takes](https://github.com/dimagi/commcare-hq/blob/432359e491a4ff2e07474c752cb7c2188447e0bf/corehq/apps/domain/static/domain/js/billing_statements.js#L22). But it repeats code, and it seems very clumsy for adding just one function.
* Add the function to the viewmodel instance. @proteusvacuum it looks like you did something similar in [incremental_export.js](https://github.com/dimagi/commcare-hq/blob/6c5bfec635b9dffaae45f50af34ef019eece4bde/corehq/apps/export/static/export/js/incremental_export.js#L8). But my attempt to do this for `sendNow` did not work. It just creates a new instance.

Is there a clean and simple way to add `sendNow` to the viewmodel? Or should I just leave it as a global?

Reply:
(To be followed up in a future PR.)

Reply:
Say more about the creation of a new instance? I don't see why what I did in `incremental_export` wouldn't work here.